tests/websocket_server.py

```diff
@@ -0,0 +1,60 @@
+import threading
+
+import pytest
+from ws4py.server.wsgirefserver import WebSocketWSGIRequestHandler, WSGIServer
+from ws4py.server.wsgiutils import WebSocketWSGIApplication
+from wsgiref.simple_server import make_server
+
+
+class WebsocketServer(threading.Thread):
```

Comment:
I'm afraid I really think the threads in the tests should go. We'll come up with a solution for the IO code later.